### PR TITLE
issue 355.3: allow IResearchLink creation, as opposed to IResearch View update, to direct which CIDs get included in an IResearchView snapshot

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

File: arangod/IResearch/IResearchView.cpp [hunk 1]
```diff
@@ -1573,78 +1743,40 @@ arangodb::Result IResearchView::link(
...
-    LOG_TOPIC(WARN, iresearch::IResearchFeature::IRESEARCH)
+  if (!impl._meta.init(properties, error)) {
+    LOG_TOPIC(WARN, IResearchFeature::IRESEARCH)
       << "failed to initialize iResearch view from definition, error: " << error;
 
     return nullptr;
   }
 
-  return view;
+  return std::move(view);
```
Comment (gnusi) [inline arangod/IResearch/IResearchView.cpp]: do not use explicit move, it implicitly disable RVO

File: arangod/VocBase/LogicalView.h [hunk 2]
```diff
@@ -163,10 +163,26 @@ class DBServerLogicalView : public LogicalView {
     bool isNew
   );
 
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief fill and return a jSON description of a View object implementation
+  //////////////////////////////////////////////////////////////////////////////
+  virtual void getPropertiesVPack(
```
Comment (gnusi) [inline arangod/VocBase/LogicalView.h]: why do we need this function?

File: arangod/VocBase/LogicalView.h [hunk 3]
```diff
@@ -163,10 +163,26 @@ class DBServerLogicalView : public LogicalView {
...
+    velocypack::Builder& builder,
+    bool forPersistence
+  ) const = 0;
+
+  ///////////////////////////////////////////////////////////////////////////////
+  /// @brief called when a view's properties are updated (i.e. delta-modified)
+  ///////////////////////////////////////////////////////////////////////////////
+  virtual arangodb::Result updateProperties(
```
Comment (gnusi) [inline arangod/VocBase/LogicalView.h]: why do we need this function?

File: arangod/IResearch/IResearchView.cpp (hunk 1 above)
Comment (vasiliy-arangodb) [reply arangod/IResearch/IResearchView.cpp]: this code was there for a quite long while

File: arangod/VocBase/LogicalView.h (hunk 2 above)
Comment (vasiliy-arangodb) [reply arangod/VocBase/LogicalView.h]: to fill and return a jSON description of a View object implementation

File: arangod/VocBase/LogicalView.h (hunk 3 above)
Comment (vasiliy-arangodb) [reply arangod/VocBase/LogicalView.h]: it is called when a view's properties are updated (i.e. delta-modified)

File: arangod/IResearch/IResearchView.cpp (hunk 1 above)
Comment (graetzer) [reply arangod/IResearch/IResearchView.cpp]: You guys should compile with clang more often, I get quite a few warnings about places with `return std::move(...)`

**Pull request merged**